src/services/data_model_engine.js

```diff
@@ -103,10 +104,25 @@ export default class DataModelEngine {
     return asset;
   }
 
+  async selectAssetsByIdentifier(identifiers, accessLevel) {
+    const findEventQueryObject = this.findEventQueryObjectFactory.create({
+      data: {
+        type: 'ambrosus.event.identifiers',
+        ...identifiers
+      }
+    }, accessLevel ? accessLevel : 0);
+    const events = await findEventQueryObject.execute();
+    return [...new Set(events.results.map((event) => event.content.idData.assetId))];
+  }
+
   async findAssets(params, tokenData) {
-    const validatedParams = this.entityBuilder.validateAndCastFindAssetsParams(params);
-    const accessLevel = await this.accountAccessDefinitions.getTokenCreatorAccessLevel(tokenData);
-    const findAssetQueryObject = this.findAssetQueryObjectFactory.create(validatedParams, accessLevel);
+    let validatedParams = this.entityBuilder.validateAndCastFindAssetsParams(params);
+    if (validatedParams.identifier) {
+      const accessLevel = await this.accountAccessDefinitions.getTokenCreatorAccessLevel(tokenData);
+      const consideredAssets = await this.selectAssetsByIdentifier(validatedParams.identifier, accessLevel);
+      validatedParams = pick(put(validatedParams, 'consideredAssets', consideredAssets), 'identifier');
```

Comment:
I would change `consideredAssets` to just `assetIds`. In the future we could then introduce direct querying of multiple assets by theirs assetsIds.